src/m365/commands/setup.ts

```diff
@@ -106,59 +148,215 @@ class SetupCommand extends AnonymousCommand {
+    }
+
+    await logger.logToStderr('');
+    await logger.logToStderr('Configuring settings...');
+    await logger.logToStderr('');
+
+    await this.configureSettings({ preferences, settings, silent: false, logger });
+
+    if (!this.verbose) {
       await logger.logToStderr('');
-      await logger.logToStderr('Configuring settings...');
-      await logger.logToStderr('');
+      await logger.logToStderr(chalk.green('DONE'));
+    }
+  }
 
-      await this.configureSettings(settings, false, logger);
+  private async configureExistingEntraApp(logger: Logger): Promise<Preferences> {
+    await logger.logToStderr('Please provide the details of the existing app registration.');
+    let clientCertificateFile: string | undefined;
+    let clientCertificateBase64Encoded: string | undefined;
+    let clientCertificatePassword: string | undefined;
 
-      if (!this.verbose) {
-        await logger.logToStderr('');
-        await logger.logToStderr(chalk.green('DONE'));
+    const clientId = await cli.promptForInput({
+      message: 'Client ID:',
+      /* c8 ignore next */
+      validate: value => validation.isValidGuid(value) ? true : 'The specified value is not a valid GUID'
+    });
+    const tenantId = await cli.promptForInput({
+      message: 'Tenant ID (leave common if the app is multitenant):',
+      default: 'common',
+      /* c8 ignore next */
+      validate: value => value === 'common' || validation.isValidGuid(value) ? true : `Tenant ID must be a valid GUID or 'common'`
+    });
+    const clientSecret = await cli.promptForInput({
+      message: 'Client secret (leave empty if you use a certificate or a public client):'
+    });
+    if (!clientSecret) {
+      clientCertificateFile = await cli.promptForInput({
+        message: `Path to the client certificate file (leave empty if you want to specify a base64-encoded certificate string):`
+      });
+      if (!clientCertificateFile) {
+        clientCertificateBase64Encoded = await cli.promptForInput({
+          message: `Base64-encoded certificate string (leave empty if you don't connect using a certificate):`
+        });
+      }
+      if (clientCertificateFile || clientCertificateBase64Encoded) {
+        clientCertificatePassword = await cli.promptForInput({
+          message: 'Password for the client certificate (leave empty if the certificate is not password-protected):'
+        });
       }
     }
+
+    return {
+      clientId,
+      tenantId,
+      clientSecret,
+      clientCertificateFile,
+      clientCertificateBase64Encoded,
+      clientCertificatePassword
+    };
   }
 
-  private getSummaryMessage(settings: SettingNames): string {
+  private async createNewEntraApp(preferences: Preferences, logger: Logger): Promise<AppInfo> {
+    if (!await cli.promptForConfirmation({
+      message: 'CLI for Microsoft 365 will now sign in to your Microsoft 365 tenant as Microsoft Azure CLI to create a new app registration. Continue?',
+      default: false
```

Comment:
I chose to default to false so that it's an explicit decision for you to let us create an Entra app in your tenant. Registering an app is intrusive and I'd like the user to be sure they understand it and are ok with it.